src/functions/indBinary.jl

```diff
@@ -13,7 +13,7 @@ S = \\{ x : x_i = low_i\\ \\text{or}\\ x_i = up_i \\},
 ```
 Parameters `low` and `up` can be either scalars or arrays of the same dimension as the space.
 """
-struct IndBinary{C <: RealOrComplex, T <: Union{C, AbstractArray{C}}, S <: Union{C, AbstractArray{C}}} <: ProximableFunction
```

Comment:
wow this was strict!

Reply:
yes :-) there's a lot of overly strict constraints, I fixed a few for now that are necessary to have tests pass on 1.6, we'll fix more later